src/ebuckets.c

```diff
@@ -0,0 +1,2244 @@
+            ebDestroy(&eb, &myEbucketsType2, NULL);
+        }
+    }
+
+    TEST_COND("ebuckets - test min/max expire time, with extended-segment",
+              (1<<EB_BUCKET_KEY_PRECISION) > 2*EB_SEG_MAX_ITEMS) {
+        ebuckets eb = NULL;
+        MyItem items[(2*EB_SEG_MAX_ITEMS)-1];
+        for (int numItems = EB_SEG_MAX_ITEMS+1 ; numItems < (int)ARRAY_SIZE(items) ; numItems++) {
+            /* First reach extended-segment (two chained segments in a bucket) */
+            for (int i = 0; i <= EB_SEG_MAX_ITEMS; i++) {
+                uint64_t itemExpireTime = (1<<EB_BUCKET_KEY_PRECISION) + i;
+                ebAdd(&eb, &myEbucketsType2, items + i, itemExpireTime);
+            }
+
+            /* Now start adding more items to extended-segment and verify min/max */
+            for (int i = EB_SEG_MAX_ITEMS+1; i < numItems; i++) {
+                uint64_t itemExpireTime = (1<<EB_BUCKET_KEY_PRECISION) + i;
+                ebAdd(&eb, &myEbucketsType2, items + i, itemExpireTime);
+                assert(ebGetNextTimeToExpire(eb, &myEbucketsType2) == (uint64_t)(2<<EB_BUCKET_KEY_PRECISION));
+                assert(ebGetMaxExpireTime(eb, &myEbucketsType2, 0) == (uint64_t)(2<<EB_BUCKET_KEY_PRECISION));
+                assert(ebGetMaxExpireTime(eb, &myEbucketsType2, 1) == (uint64_t)((1<<EB_BUCKET_KEY_PRECISION) + i));
+            }
+            ebDestroy(&eb, &myEbucketsType2, NULL);
+        }
+    }
+
+    TEST("ebuckets - active-expire dry-run") {
+        ebuckets eb = NULL;
+        MyItem items[2*EB_SEG_MAX_ITEMS];
+
+        for (int numItems = 1 ; numItems < (int)ARRAY_SIZE(items) ; numItems++) {
+            int maxExpireKey = (numItems % 2) ? 40 : 2;
+            /* Allocate numItems and add to ebuckets */
+            for (int i = 0; i < numItems; i++) {
+                /* generate random expiration time */
+                uint64_t expireTime = (rand() % maxExpireKey) << EB_BUCKET_KEY_PRECISION;
+                ebAdd(&eb, &myEbucketsType2, items + i, expireTime);
+            }
+
+            for (int i = 0 ; i <= maxExpireKey ; ++i) {
+                uint64_t now = i << EB_BUCKET_KEY_PRECISION;
+
+                /* Count how much items are expired */
+                uint64_t expectedNumExpired = 0;
+                for (int j = 0; j < numItems; j++) {
+                    if (ebGetExpireTime(&myEbucketsType2, items + j) < now)
+                        expectedNumExpired++;
+                }
+                /* Perform dry-run and verify number of expired items */
+                assert(ebExpireDryRun(eb, &myEbucketsType2, now) == expectedNumExpired);
+            }
+            ebDestroy(&eb, &myEbucketsType2, NULL);
+        }
+    }
+
+//    TEST("segment - Add smaller item to full segment that all share same ebucket-key")
+//    TEST("segment - Add item to full segment and make it extended-segment (all share same ebucket-key)")
+//    TEST("ebuckets - Create rax tree with extended-segment and add item before")
+
```

Comment:
```suggestion
```

Reply:
it is a placeholder for missing tests. i perfer to leave it as is if you don't mind.

Reply:
i perfer to add a todo.